src/main/java/com/graphhopper/converter/resources/ConverterResourceGisgraphy.java

```diff
@@ -0,0 +1,185 @@
+        String lat = null;
+        String lng = null;
+        if (point!=null && !point.isEmpty() && point.indexOf(",")>0){
+            String[] cords = point.split(",");
+            lat = cords[0];
+            lng = cords[1];
+        }
+
+
+        WebTarget target;
+        if (reverse) {
+            target = buildReverseGeocodingTarget(query, lat, lng, radius,
+                    country, limit);
+        } else if (autocomplete) {
+            target = buildAutocompleteTarget(query, lat, lng, radius, country,
+                    limit);
+        } else {
+            target = buildGeocodingTarget(query, lat, lng, radius, country,
+                    limit);
+        }
+
+        Response response = target.request().accept("application/json").get();
+        Status status = new Status(response.getStatus(), response
+                .getStatusInfo().getReasonPhrase());
+        failIfResponseNotSuccessful(target, status);
+
+
+        try {
+            if (!autocomplete) {
+                GisgraphyGeocodingResult feed = response
+                        .readEntity(GisgraphyGeocodingResult.class);
+                return Converter.convertFromGisgraphyList(feed.result, status);
+            }
+            else {
+                GisgraphySearchResult feed = response
+                        .readEntity(GisgraphySearchResult.class);
+                return Converter.convertFromGisgraphySearchList(feed.getResponse(), status);
+            }
+        } catch (Exception e) {
+            LOGGER.error("There was an issue with the target "
+                    + target.getUri() + " the provider returned: "
+                    + status.code + " - " + status.message);
+            throw new BadRequestException(
+                    "error deserializing geocoding feed");
+        }
+    }
+
+
+    private void checkParameters(String query, boolean reverse, boolean autocomplete, String point) {
+        super.checkInvalidParameter(reverse,query,point);
+        if (reverse && autocomplete){
+            throw new BadRequestException("autocomplete is not available in reverse geocoding request, set reverse or autocomplete to false but not both");
+        }
+
+    }
+
+    private WebTarget buildGeocodingTarget(String query, String lat,
+            String lng, String radius, String country, int limit) {
+        WebTarget target = buildBaseTarget(lat, lng, geocodingUrl)
+                .queryParam(ADDRESS_PARAMETER, query);;
+                target = addRadiusAndCountryToTarget(radius, country, target);
+                if (limit >0) {
+                    target = target.queryParam(GEOCODING_LIMIT_PARAMETER, limit);
+                }
+                return target;
+    }
+
+
+
+    private WebTarget buildAutocompleteTarget(String query, String lat,
+            String lng, String radius, String country, int limit) {
+        WebTarget target = buildBaseTarget(lat, lng, searchURL);
+        target.queryParam(SEARCH_QUERY_PARAMETER, query)
```

Comment:
Here a `target = ` is missing or we have to remove the `;` in the line above, also see the failing test.